app/src/main/java/com/gdsctsec/smartt/viewmodel/HomeScreenViewModel.kt

```diff
@@ -8,10 +8,11 @@ import com.gdsctsec.smartt.data.Weekday
 import com.gdsctsec.smartt.data.repository.LectureRepository
 import java.util.*
 
-class HomeScreenViewModel(context: Context) : ViewModel() {
+class HomeScreenViewModel(private val context: Context) : ViewModel() {
 
     private lateinit var lecturesOfTheDayLiveList: LiveData<List<TimeTable>>
-
+    private lateinit var weekday: Weekday
+    private lateinit var monthDate: String
     init {
         lecturesOfTheDayLiveList =
             LectureRepository(context, getWeekday()).getLecturesByWeekday(getWeekday())
```

Comment:
where is the top level reference for repository? I had said for repository! and not for weekday and monthDate